src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -513,6 +515,14 @@ DatabaseShardImp::setStored(std::shared_ptr<Ledger const> const& ledger)
         complete_.emplace(incomplete_->index(), std::move(incomplete_));
         incomplete_.reset();
         updateStats(l);
+
+        // Update peers with new shard index
+        protocol::TMShardInfo message;
+        auto const& publicKey {app_.nodeIdentity().first};
+        message.set_nodepubkey(publicKey.data(), publicKey.size());
+        message.set_shardindexes(std::to_string(shardIndex));
+        app_.overlay().foreach(send_always(
+            std::make_shared<Message>(message, protocol::mtSHARD_INFO)));
```

Comment:
We can't arbitrarily send this message out to other peers; if they don't support `mtSHARD_INFO` it will result in the closing their connection to this server.

We need to know if a peer supports sharding _before_ we send them shard-related messages.

Reply:
@nbougalis I can't find the code that disconnects upon receiving an unknown message and I've been unable to reproduce that behavior using the tip of develop. `Peerimp::onMessageUnknown` is called by `invokeProtocolMessage` when an unknown message is received but the function is just a stub with a `TODO`. It seems someone intended on adding the behavior or I did I miss something else entirely?